src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -3856,6 +3862,7 @@ internal sealed class TestingStuff
             internal Action RestoreDatabaseAfterSavingDatabaseRecord;
             internal Action AfterCommitInClusterTransaction;
             internal Action<string, List<ClusterTransactionCommand.SingleClusterDatabaseCommand>> BeforeExecuteClusterTransactionBatch;
+            internal Func<CommandBase, long, long> ModifyTermBeforeRachisMergedCommandInsertToLeaderLog;  // gets: command and term, returns: new term
```

Comment:
better put this in the `TestingStuff` of `RachisConsensus`

Reply:
RachisConsensus.TestingStuff has implemented methods (not Func/Action), so if you create a TestingStuff for RachisConsensus (not null) - the methods are automatically invoked.

Reply:
So we need to adjust it.. because it is the proper place for it.

Reply:
How do you want to adjust it? 
It has implemented methods that are invoked once you create the `RachisConsensus.TestingStuff` object, I don't see how to do it without breaking other tests..
I don't see any other way except creating an object that inherits from `RachisConsensus.TestingStuff` in my test and overriding the methods with "empty methods", which is not a very pretty solution.

Reply:
yes, you will have those invocations which all seems to be related to `HoldOnLeaderElect`, so we can put it all under some class? 
I'm sure we can do something here..